src/safemode_ui.h

```diff
@@ -0,0 +1,124 @@
+#ifndef SEFEMODE_UI_H
+#define SEFEMODE_UI_H
+
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <locale>
+#include <algorithm>
+#include "json.h"
+#include "enums.h"
+#include "creature.h"
+
+class safemode : public JsonSerializer, public JsonDeserializer
+{
+    private:
+        enum Tabs : int {
+            GLOBAL_TAB,
+            CHARACTER_TAB,
+            MAX_TAB
+        };
+
+        class rules_class
+        {
+            public:
+                std::string rule;
+                bool active;
+                bool exclude;
+                Creature::Attitude attitude;
+                int proximity;
+
+                rules_class() {
```

Comment:
Besides that `rule = "";` is pointless thing at this point (`rule` is a string, which is automatically empty at this point), you should use member initializer list:
```C++
rules_class() : rule(), active(false), exclude(false), attitude(...), proximity(...) {
}
```
(`rule()` here is optional, its default constructor will be called if it's not listed)

This avoids all the `this->`. Same should be done with the other constructor.